test/interpreter/function-sumif.spec.ts

```diff
@@ -452,6 +452,71 @@ describe('Function SUMIF(S) - calculations and optimizations', () => {
 
     expect(engine.getCellValue(adr('A5'))).toEqual(13)
   })
+
+  it('coerces dates as numbers', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['1', '9160250011660588', '43469', '25000'],
+      ['2', '9160250011689568', '43631', '15000'],
+      ['=SUMIF(C2:C11,">31/05/2019",D2:D11)']
+    ], {dateFormats: ['DD/MM/YYYY']})
+    expect(engine.getCellValue(adr('A3'))).toEqual(15000)
+  })
+
+  it('works when criterion arg is a date', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['31/05/2019', '1'],
+      ['=DATEVALUE("31/05/2019")', '1'],
+      ['=DATE(2019, 5, 31)', '1'],
+      ['=A1', '1'],
+      ['43616', '1'],
+      ['=SUMIF(A1:A5, "31/05/2019", B1:B5)']
+    ], {dateFormats: ['DD/MM/YYYY']})
+
+    expect(engine.getCellValue(adr('A6'))).toEqual(5)
+  })
+
+  it('works when criterion arg is a time value', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['12:13', '1'],
+      ['=SUMIF(A1:A1, "12:13", B1:B1)']
+    ])
+
+    expect(engine.getCellValueDetailedType(adr('A1'))).toEqual(CellValueDetailedType.NUMBER_TIME)
+    expect(engine.getCellValue(adr('A2'))).toEqual(1)
+  })
+
+  it('works when criterion arg is a datetime value', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['31/05/2019 9:15', '1'],
+      ['=SUMIF(A1:A1, "31/05/2019 9:15", B1:B1)']
+    ])
+
+    expect(engine.getCellValueDetailedType(adr('A1'))).toEqual(CellValueDetailedType.NUMBER_DATETIME)
+    expect(engine.getCellValue(adr('A2'))).toEqual(1)
+  })
+
+  it('works when criterion arg is a currency value', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['$1', '1'],
+      ['1$', '1'],
+      ['=SUMIF(A1:A2, "$1", B1:B2)'],
+      ['=SUMIF(A1:A2, "1$", B1:B2)']
```

Comment:
Because of the text trimming, the following criterions also succeed:

```
      ['=SUMIF(A1:A2, "  $1", B1:B2)'], // start with whitespace
      ['=SUMIF(A1:A2, "1$  ", B1:B2)'] // end with whitespace
```

However:
1. is that according to the spec?
1. if yes, then a test for it is missing
1. if yes, then it seems that Excel has slightly different implementation. In my Excel, start with whitespace succeeds, but end with whitespace fails.

Reply:
I [investigated it](https://github.com/handsontable/hyperformula/pull/1021#issuecomment-1195773214) and added a test to make sure whitespace characters are ignored when evaluating the criterion string.